libvirt/tests/src/libvirt_hooks.py

```
@@ -298,21 +300,77 @@ def network_hook():
             assert check_hooks(hook_str)
             hook_str = hook_file + " " + net_name + " started begin -"
             assert check_hooks(hook_str)
+            if vm.is_alive():
+                vm.destroy(gracefully=False)
 
-            # plug a interface
-            if vm.is_dead():
-                vm.start()
-                vm.wait_for_login().close()
+            # Remove all controllers, interfaces and addresses in vm dumpxml
+            vm_inactive_xml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+            vm_inactive_xml.remove_all_device_by_type('controller')
+            type_dict = {'address': '/devices/*/address'}
+            try:
+                for elem in vm_inactive_xml.xmltreefile.findall(type_dict['address']):
+                    vm_inactive_xml.xmltreefile.remove(elem)
+            except (AttributeError, TypeError) as details:
+                logging.error("Fail to remove address: %s", details)
+                return False
```

I assume you want to break the execution here, so suggest use test.fail(xx) rather than return false

---

will update